src/crimson/os/cyanstore/cyan_store.cc

```diff
@@ -55,16 +55,24 @@ struct singleton_ec : std::error_code {
 
 seastar::future<> CyanStore::start()
 {
-  return shard_stores.start(this);
+  return shard_stores.start(path);
 }
 
-seastar::future<store_statfs_t> CyanStore::stat() const
+seastar::future<store_statfs_t> CyanStore::stat()
 {
   logger().debug("{}", __func__);
-  store_statfs_t st;
-  st.total = crimson::common::local_conf().get_val<Option::size_t>("memstore_device_bytes");
-  st.available = st.total - used_bytes;
-  return seastar::make_ready_future<store_statfs_t>(std::move(st));
+  return shard_stores.map([](auto &local_store) {
```

Comment:
Looks seastar has already provided the const version of `get_local_shard()` used by `map()`:
https://github.com/scylladb/seastar/blob/3db15b5681c460a759641a3907eb9db84bdcf395/include/seastar/core/sharded.hh#L514-L520

Does this work instead of dropping `const` from `FuturizedStore::stat()`?
```c++
return shard_stores.map([](const CyanStore::ShardStores &local_store) {
```

Reply:
doesn't work. @cyx1231st

Reply:
How about replacing this commit by https://github.com/cyx1231st/ceph/commit/0c7ec5681cceb4415465b14779854c2f925205b5 ?